### ISSUE TITLE
Visualize branching and merging history

### ISSUE THREAD

**Issue by borekb** [versionpress/versionpress#1043]
We could add a panel that would visualize the branching and merging history of the site, perhaps with the summary of commits that happened there (maybe optionally, to keep the UI simple). Something like this:

```
                          ^
             PRODUCTION   |    STAGING
                          |
                          +-------+
                          |       |
Added Production post *   |       |   * Added About Us page
    Edited option xyz *   |       |   * Customized theme
                          |       |   * Changed theme
                          |       |
                          +-------+
                          |
   Updated site title *   |
  Trashed Hello World *   |
         Installed VP *   |
                          +
```

In the future, we could also consider SourceTree-like visualization of commit graph right to the main table but maybe it would be just a clutter for common non-technical users

![2016-05-11 15_40_41-clipboard](https://cloud.githubusercontent.com/assets/101152/15182860/d2ab3c12-178e-11e6-989a-e81524572921.png)

**Comment by borekb** [comment]
We discussed some of the specifics today with @Ishirak, @JanVoracek and @octopuss. This feature could be quite big and unwieldy one day, however, let's start small and expand in the future.

Some of the core decisions:

- We'll start with **rendering the current page of commits only**. For example, if I'm on page 1 our of 7, the graph will only visualize the page 1.
    - Technically, this means that the graph can easily be rendered client-side. If we were to render the full history, it would not be efficient to download hundreds of commits only to show a couple of branching and merging points.
- There might be "load more" or "view the full history" switches later, however, that might require new server-side endpoints as per the previous bullet point.

Regarding the UI:

- The core is to be able to draw the "rails". Whether we'll display commit details or not is secondary.
- Let's develop the UI as a separate component / panel first, but we also discussed how the visualization could perhaps be a watermark under the main table, or something like that.

**Comment by borekb** [comment]
:four_leaf_clover: Shipped in 4.0-alpha1 at this point. Keeping this issue open for future updates until 4.0 is final.

**Comment by borekb** [comment]
Taking it back: let's do it as with #1036, i.e., close the original issue and create smaller issue for post-alpha improvements.

**Comment by borekb** [comment]
:ship: Shipped in [4.0-alpha1](https://github.com/versionpress/versionpress/releases/tag/4.0-alpha1).
